vaadin-date-time-picker-flow-integration-tests/src/test/java/com/vaadin/flow/component/datetimepicker/DateTimePickerIT.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2000-2019 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.datetimepicker;
+
+import java.time.LocalDateTime;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.By;
+
+import com.vaadin.flow.component.datetimepicker.demo.DateTimePickerView;
+import com.vaadin.flow.component.datetimepicker.testbench.DateTimePickerElement;
+import com.vaadin.flow.testutil.AbstractComponentIT;
+import com.vaadin.flow.testutil.TestPath;
+import com.vaadin.testbench.TestBenchElement;
+
+/**
+ * Integration tests for the {@link DateTimePickerView}.
+ */
+@TestPath("date-time-picker-it")
+public class DateTimePickerIT extends AbstractComponentIT {
+
+    private DateTimePickerElement picker;
+    private TestBenchElement message;
+
+    @Before
+    public void init() {
+        open();
+        waitForElementPresent(By.tagName("vaadin-date-time-picker"));
+        picker = $(DateTimePickerElement.class).first();
+        message = $("div").id("message");
+    }
+
+    @Test
+    public void selectDateTime() {
+        picker.setDateTime(LocalDateTime.of(1985, 1, 10, 13, 51));
+
+        waitUntil(driver -> message.getText().contains("1985-01-10 13:51:00"));
```

Comment:
You should be able to use `Assert`. In a normal case, TestBench waits for the server-roundtrip automatically. `waitUntil` can be used for assertion when there's some extra asynchronousity which is not handled automatically. If it fails, it prints a generic `TimeoutException`.

You could use `Assert.assertTrue()` or even better:
```java
Assert.assertThat(message.getText(), CoreMatchers.containsString("1985-01-10 13:51:00"));
```
This method knows how to build an useful error message if it fails. Like "expected the text to contain X but the text was Y".

All the assert methods can also take custom error message as the first parameter. It depends on the case if it should be used. With `CoreMatchers.containsString` it's not maybe necessary as it provides a really useful message already. With `assertTrue` it would be beneficial to provide a better reason for the failure than the default "expected false to be true".

Reply:
I was thinking about this, but didn't yet try removing the `waitUntil` (it was inherited from DatePicker tests). Thanks for the hint about `assertThat` and `CoreMatchers`!

I'll test if using this approach works.